elastic/elastic-agent
# Fix log level value reported by Agent to Fleet

## What does this PR do?

This PR fixes the log level reported by Agent when checking in with Fleet. 
Since merging #3090, a managed elastic-agent has now 2 sources of configuration for log level:
1. agent-specific log level setting saved in `fleet.enc`
2. log level setting in the Fleet policy itself (with lower priority than option 1.)

Before this change, the agent was sending only the information coming from 1. (with a default value equal to `info`), now we send back to Fleet the actual log level specified in coordinator state (which is the end result of evaluating the multiple settings with the correct priority)

There has also been a modification of the integration test related to the feature with additional assertions on log level received by Fleet.



## Why is it important?

The log level is now correctly displayed in Fleet UI regardless for each agent, regardless of how it's been set (agent settings, policy or default value)



## Checklist




## Disruptive User Impact



## How to test this PR locally



1. Spin up a Stack deployment
2. Create a policy without any specific log level settings (check in `Advanced settings` under the Policy's `settings` Tab  
  ![image](https://github.com/elastic/elastic-agent/assets/96178987/1bcacadd-0beb-421a-a283-edb8ea48529d)
3. Install and enroll an agent and verify that the default log level reported is `info`
4. Set some log level in policy and check in the agent details that the change is reported correctly (Ignore the `offline` state in the image below, the screenshot was taken after the agent has been uninstalled)
  ![image](https://github.com/elastic/elastic-agent/assets/96178987/d2377fe9-9ff3-46cf-92ef-913f6b299d6e)
5. Set some specific (different!) log level in the agent settings tab and check that in a few seconds that change is also reflected in UI
6. Have fun! Try any combination of setting/unsetting the log level and verify that the agent log level is updated within a reasonable time in the agent details tab

## Related issues


- Closes #4747

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
